crates/subspace-service/src/lib.rs

```diff
@@ -108,6 +110,37 @@ pub type FullClient<RuntimeApi, ExecutorDispatch> =
 pub type FullBackend = sc_service::TFullBackend<Block>;
 pub type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
 
+/// Subspace-specific service configuration.
+pub struct SubspaceConfiguration {
+    base: Configuration,
+}
+
+impl From<Configuration> for SubspaceConfiguration {
+    fn from(base: Configuration) -> Self {
+        Self { base }
+    }
+}
+
+impl From<SubspaceConfiguration> for Configuration {
+    fn from(configuration: SubspaceConfiguration) -> Self {
+        configuration.base
+    }
+}
+
+impl Deref for SubspaceConfiguration {
+    type Target = Configuration;
+
+    fn deref(&self) -> &Self::Target {
+        &self.base
+    }
+}
+
+impl DerefMut for SubspaceConfiguration {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.base
+    }
+}
```

Comment:
I guess we can use [`derive_more`](https://crates.io/crates/derive_more) for these kind of trait impls. They are very generic.

Reply:
Will use it in upcoming PR where I add more fields to it

Reply:
However worth remembering that ever derive adds to build times

Reply:
Yeah, right. But it is always a tradeoff between reviewer time and compilation time. IMO it won't add that much time.

Btw, we can generate compilation reports and see the compilation bottlenecks :)

Reply:
RocksDB is unbeatable, believe me :joy: